### PR TITLE
BI-2122 - Workflows Select

### PR DESCRIPTION
**Pull request by nickpalladino**
# Description
**Story:** [BI-2122](https://breedinginsight.atlassian.net/browse/BI-2122?atlOrigin=eyJpIjoiZmI1ODZlOWU0ODVkNDdhOGE0ZGY0MzhiODU5ZTVmNDEiLCJwIjoiaiJ9)

- Add workflow configuration to db
- Add endpoint to retrieve workflows for given import mapping
- Add endpoints to pass selected workflow id to processors
- Create WorkflowFactory to give Workflow based on workflow id

# Dependencies
- need to run new db migration
- bi-web feature/BI-2121

# Testing
- Verify that the experiment import works the same as it has prior to this card regardless of what workflow is selected (just spot check, if it runs)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have tested that my code works with both the brapi-java-server and BreedBase
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<please include a link to TAF run>_

[BI-2122]: https://breedinginsight.atlassian.net/browse/BI-2122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mlm483 — APPROVED** [review]

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+@Primary
+@Singleton
+public class ExperimentWorkflowNavigator implements ExperimentWorkflow {
+    private final List<ExperimentWorkflow> workflows;
+
+    public ExperimentWorkflowNavigator(List<ExperimentWorkflow> workflows) {
+        this.workflows = workflows;
+    }
+
```
Comment (mlm483) [inline src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java:22]: @dmeidlin @nickpalladino Is `workflows` injected in the constructor? If yes, how does that work? Usually we use the `@Inject` annotation, which is what I would have expected.

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Primary
+@Singleton
+public class ExperimentWorkflowNavigator implements ExperimentWorkflow {
+    private final List<ExperimentWorkflow> workflows;
```
Comment (mlm483) [inline src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java:17]: It seems like this is the class that orchestrates the running of the workflows, having a high-level explanation in a comment might help other developers understand the architecture.

**Review by mlm483 — APPROVED** [review]

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java (hunk 2 above)
Comment (dmeidlin) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java:17]: I've added overview comments in the WorkflowNavigator to be merged in with [BI-2134](https://github.com/Breeding-Insight/bi-api/pull/359/files#diff-83f2deb13f2ea62ba49ed53c10bf4bc30ccf63cf6105a130a9c9056597f2a5c3).

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java (hunk 1 above)
Comment (dmeidlin) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java:22]: ExperimentWorkflow is a @FunctionalInterface. Micronaut scans the code base and collects all instances of ExperimentWorkflow into an ordered list. This list is automatically available for constructor injection for the instance of ExperimentWorkflow annotated with @Primary.

**Review by dmeidlin — APPROVED** [review]

**Pull request merged**